### PR TITLE
[CAMEL-21381] camel k8s cannot run spring-boot on openshift

### PR DESCRIPTION
**Pull request by tdiesler**
https://issues.apache.org/jira/browse/CAMEL-21381

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by davsclaus — APPROVED** [review]

File: dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/test/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/PodLogsTest.java [hunk 1]
```diff
@@ -33,7 +33,8 @@ public void shouldHandlePodNotFound() throws Exception {
         PodLogs command = createCommand();
         command.name = "mickey-mouse";
         command.maxWaitAttempts = 2; // total timeout of 4 seconds
-        command.doCall();
+        int exit = command.doCall();
+        Assertions.assertEquals(1, exit);
```
Comment (davsclaus) [inline dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/test/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/PodLogsTest.java:37]: Just wonder why exit code 1 is expected, is the command supposed to fail somehow ? If so maybe add a code comment to tell this reason so we wont get confused later looking back at this code

Comment (tdiesler) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/test/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/PodLogsTest.java:37]: It is debatable, wether the `jbang camel kubernetes logs ...` command should "fail" when the Pod is not there. User feedback (in print the output) is clear. We might as well succeed the command and reserve failure to unexpected server failures. Would that be better?

Comment (davsclaus) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/test/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/PodLogsTest.java:37]: Yes I think this is better

Comment (tdiesler) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/test/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/PodLogsTest.java]: Done

**Comment by tdiesler** [comment]
Thanks for merging. Could you perhaps leave the cherry-pick to 4.8.x to me - I'd like to run some (manual) tests before we commit to that otherwise stable branch?

**Pull request merged**